coil-svg/src/commonMain/kotlin/coil3/svg/Svg.kt

```diff
@@ -0,0 +1,59 @@
+package coil3.svg
+
+import coil3.Image
+import coil3.annotation.ExperimentalCoilApi
+import coil3.request.Options
+import coil3.svg.internal.parseSvg
+import kotlin.jvm.JvmField
+import okio.BufferedSource
+
+/**
+ * Represents an SVG with mutable properties.
+ *
+ * NOTE: This interface is provided to enable custom SVG parsing/rendering implementations, however
+ * it intentionally only provides hooks for the properties Coil uses. Additionally, these classes
+ * are very likely to change in the future.
+ */
+@ExperimentalCoilApi
+interface Svg {
```

Comment:
I'm not stoked about the API on this class (especially the getters and setters), but it should do the job for now.